### PR TITLE
[DOMA-3611] 🧭 Meter app linking 🧭

### PR DESCRIPTION
**Pull request by SavelevMatthew**
image
image

### CODE REVIEW

File: apps/condo/domains/meter/schema/Meter.js [hunk 1]
```diff
@@ -165,6 +166,22 @@ const Meter = new GQLListSchema('Meter', {
             isRequired: true,
             defaultValue: false,
         },
+        b2cApp: {
```
Comment (pahaz) [inline apps/condo/domains/meter/schema/Meter.js:169]: We need to restrict access to update/create this field

Comment (SavelevMatthew) [reply apps/condo/domains/meter/schema/Meter.js:169]: miniapps use OIDC authorization to perform actions "on behalf of the user". Therefore, I still have no idea how you can cut the rights in this place.

File: apps/condo/domains/common/components/containers/FormList.tsx [hunk 2]
```diff
@@ -429,18 +432,21 @@ const BaseModalForm: React.FC<IBaseModalFormProps> = ({
...
-                        <>
-                            {typeof children === 'function' ? children(form) : children}
-                        </>
-                    )
+        <Space direction={'vertical'} size={40}>
+            {modalNotification}
+            <FormWithAction {...props}>
+                {
+                    ({ handleSave, form }) => {
+                        handleSaveRef.current = handleSave
```
Comment (dkoviazin) [inline apps/condo/domains/common/components/containers/FormList.tsx:440]: Looks like hack

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/common/components/containers/FormList.tsx (hunk 2 above)
Comment (SavelevMatthew) [reply apps/condo/domains/common/components/containers/FormList.tsx:440]: Maybe it is...But it's not mine....I just added `Space` here

Comment (SavelevMatthew) [reply apps/condo/domains/common/components/containers/FormList.tsx:440]: 😄

File: apps/condo/domains/meter/hooks/useUpdateMeterModal.tsx [hunk 3]
```diff
@@ -13,6 +13,7 @@ import { CustomButtonProps } from '@condo/domains/common/components/Button'
 
 import { BaseMeterModalForm } from '../components/BaseMeterModal/BaseMeterModalForm'
 import { Meter } from '../utils/clientSchema'
+import { AutoSourceAlert } from '../components/BaseMeterModal/AutoSourceAlert'
```
Comment (AleX83Xpert) [inline apps/condo/domains/meter/hooks/useUpdateMeterModal.tsx:16]:
```suggestion
import { AutoSourceAlert } from '@condo/domains/meter/components/BaseMeterModal/AutoSourceAlert'
```

**Review by AleX83Xpert — APPROVED** [review]

File: apps/condo/domains/meter/components/BaseMeterModal/AutoSourceAlert.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+export const AutoSourceAlert: React.FC<IAutoSourceAlertProps> = ({ sourceAppName }) => {
+    const intl = useIntl()
+    const Title = intl.formatMessage({ id: 'pages.condo.meter.AutoSourceAlert.title' })
+    const Message = intl.formatMessage({ id: 'pages.condo.meter.AutoSourceAlert.message' }, {
+        app: sourceAppName,
+    })
+    return (
+        <Alert
+            type={'warning'}
```
[the remark this answers is not in the document]
Comment (SavelevMatthew) [reply apps/condo/domains/meter/components/BaseMeterModal/AutoSourceAlert.tsx:17]: There's no rule about that and it's transpiled the same...so I prefer this type of props specification

**Pull request merged**